rp2040-hal/src/multicore.rs

```diff
@@ -0,0 +1,211 @@
+//! Multicore support
+// See [Chapter ?? Section ??](https://datasheets.raspberrypi.org/rp2040/rp2040_datasheet.pdf) for more details
+
+use crate::pac;
+
+#[cfg(feature = "alloc")]
+extern crate alloc;
+
+/// Errors for multicore operations.
+#[derive(Debug)]
+pub enum Error {
+    /// Operation is invalid on this core.
+    InvalidCore,
+    /// Core was unresposive to commands.
+    Unresponsive,
+}
+
+// We pass data to cores via the stack, so we read
+// the data off the stack and into parameters that
+// rust can read here. Ideally this would be a
+// #[naked] function but that is not stable yet.
+static MULTICORE_TRAMPOLINE: [u16; 2] = [
+    0xbd03, // pop {r0, r1, pc} - call wrapper (pc) with r0 and r1
+    0x46c0, // nop - pad this out to 32 bits long
+];
+static mut CORE1_STACK: [usize; 4096] = [0; 4096];
+
+#[inline(always)]
+fn install_stack_guard(stack_bottom: *mut usize) {
+    let core = unsafe { pac::CorePeripherals::steal() };
+
+    // Trap if MPU is already configured
```

Comment:
This implicitly forbids calling `spawn` more than once.
While I'm not sure if calling `spawn` multiple times is actually useful, it should at least be mentioned in the doc comment of `spawn`. Or, even better, `spawn` could return an `Error` when it's called a second time.

Reply:
spawn resets the core first, so it should be okay I think?

Reply:
Oh, sure - I didn't consider that the core reset also affects the MPU, as it's local to the core.